NU5 consensus changes (#5741)
zcash/zcash: doc/release-notes.md

- This release fixes a bug in `v4.6.0` that caused a consensus failure on the Zcash
  testnet at height `1,779,200`. For details see
  <https://github.com/zcash/zcash/commit/5990853de3e9f04f28b7a4d59ef9e549eb4b4cca>.
- There have been changes to the Halo2 proving system to improve consistency between the
  specification and the implementation, and these may break compatibility. See for example
  <https://github.com/zcash/halo2/commit/247cd620eeb434e7f86dc8579774b30384ae02b2>.
- There have been numerous changes to the Orchard circuit implementation since `v4.6.0`.
  See <https://github.com/zcash/orchard/issues?q=label%3AM-consensus-change-since-0.1.0-beta-1>
  for a complete list.
- A potential Faerie Gold vulnerability affecting the previous activation of NU5 on
  testnet and existing since `v4.6.0` has been mitigated.

NU5 Testnet Reactivation
------------------------

To support the aforementioned testnet consensus changes, the following changes are made in
`zcashd v4.7.0`:

- The consensus branch ID for NU5 is changed to `0xC2D6D0B4`.
- The protocol version indicating NU5-aware testnet nodes is set to `170050`.

> - The testnet reactivation height for NU5 is set to **1,840,000**.

Reviewer: ```suggestion
- The testnet reactivation height for NU5 is set to **1,842,420**.
```

> Testnet nodes that upgrade to `zcashd v4.7.0` prior to block height `1,840,000` will follow

Reviewer: ```suggestion
Testnet nodes that upgrade to `zcashd v4.7.0` prior to block height `1,842,420` will follow
```

the new testnet network upgrade. Testnet nodes that are running `zcashd v4.6.0-2` at that

> height will need to upgrade to `v4.7.0` and then run with `-reindex`.

Reviewer 2: ```suggestion
height will need to upgrade to `v4.7.0` and then run with `-reindex -rescan`.
```

> Mnemonic Recovery Phrases
> -------------------------

Reviewer 2: ```suggestion
Emergency Recovery Phrases
--------------------------
```

    added to the result.
- `z_viewtransaction` has been updated to include attributes that provide information
  about Orchard components of the transaction. Also, the `type` attribute for spend and
  output values has been deprecated and replaced by the `pool` attribute.
- `z_getnotescount` now also returns information for Orchard notes.
- The output format of `z_exportwallet` has been changed. The exported file now includes
  the mnemonic seed for the wallet, and HD keypaths are now exported for transparent
  addresses when available.
- The result value for `z_importviewingkey` now includes an `address_type` field that
  replaces the now-deprecated `type` key.
- `z_listunspent` has been updated to render unified addresses for Sapling and 
  Orchard outputs when those outputs are controlled by unified spending keys.  Outputs
  received by unified internal addresses do not include the `address` field.
- Legacy transparent address generation using `getnewaddress` no longer uses a
  preallocated keypool, but instead performs HD derivation from the wallet's mnemonic seed
  according to BIP 39 and BIP 44 under account ID `0x7FFFFFFF`.

> - `z_gettreestate` has been updated to include information about the orchard note

Reviewer 2: ```suggestion
- `z_gettreestate` has been updated to include information about the Orchard note
```
